cpp/src/barretenberg/plonk/proof_system/types/proof.hpp

```diff
@@ -11,6 +11,20 @@ struct proof {
     std::vector<uint8_t> proof_data;
 
     bool operator==(proof const& other) const = default;
+    void print()
+    {
+        size_t newline_counter = 0;
+        for (auto& byte : proof_data) {
+            if (newline_counter == 0 || newline_counter == 4 || newline_counter % 32 == 8) {
+                std::cerr << "\n"
```

Comment:
I think it would be better to just create a method to convert to string. Then you can use info to print it out. Direct use of std for io led to weird issues in the past

Reply:
Yeah, this is kinda crap, I just through it together when I was desperate. My dream is each flavor has its own proof struct, and then printing and serializing that that can be done in a more high-level and explicit manner. I suspect this would be an impactful change for other teams. For now I just delete my print function.